src/ui/components/DevTools.js

```diff
@@ -1,102 +1,132 @@
-const React = require("react");
+import React, { useState, useEffect } from "react";
 import { connect } from "react-redux";
 
 import Toolbox from "./Toolbox";
 import Comments from "./Comments";
 import Recordings from "./Recordings/index";
 import Tooltip from "./Tooltip";
 import Header from "./Header";
-
+import Loader from "./shared/Loader";
+import { UserPrompt } from "./Account/index";
 import SplitBox from "devtools/client/shared/components/splitter/SplitBox";
 import RightSidebar from "./RightSidebar";
+
 import { actions } from "../actions";
 import { selectors } from "../reducers";
+import { Recording } from "./Recordings/Recording";
+import { screenshotCache, nextPaintEvent, getClosestPaintPoint } from "protocol/graphics";
 
-class DevTools extends React.Component {
-  state = {
-    orientation: "bottom",
-  };
-
-  renderViewer(toolbox) {
-    const { tooltip } = this.props;
-    return (
-      <div id="outer-viewer">
-        <div id="viewer">
-          <canvas id="graphics"></canvas>
-          <div id="highlighter-root"></div>
-        </div>
-        <RightSidebar toolbox={toolbox} />
-        <Tooltip tooltip={tooltip} />
+function Viewer({ tooltip }) {
+  return (
+    <div id="outer-viewer">
+      <div id="viewer">
+        <canvas id="graphics"></canvas>
+        <div id="highlighter-root"></div>
       </div>
-    );
+      <RightSidebar />
+      <Tooltip tooltip={tooltip} />
+    </div>
+  );
+}
+
+function DevtoolsSplitBox({ updateTimelineDimensions, tooltip }) {
+  const toolbox = <Toolbox />;
+  const viewer = <Viewer tooltip={tooltip} />;
+
+  return (
+    <SplitBox
+      style={{ width: "100vw", overflow: "hidden" }}
+      splitterSize={1}
+      initialSize="50%"
+      minSize="20%"
+      maxSize="80%"
+      vert={false}
+      onMove={num => updateTimelineDimensions()}
+      startPanel={viewer}
+      endPanel={toolbox}
+      endPanelControl={false}
+    />
+  );
+}
+
+function PreviewContainer({ preview }) {
+  if (!preview) {
+    return <div className="preview-container empty" />;
   }
 
-  renderSplitBox() {
-    const { updateTimelineDimensions } = this.props;
-    const { orientation } = this.state;
+  const image = `url(data:${preview.mimeType};base64,${preview.data})`;
+  return <div className="preview-container" style={{ backgroundImage: image }}></div>;
+}
 
-    let startPanel, endPanel;
-    const vert = orientation != "bottom";
-    const toolbox = <Toolbox />;
+function RecordingLoadingScreen({ loading, recordingDuration }) {
```

Comment:
This might warrant being in its own file... 

Also i think we might want to ask apollo for the recording image, title, etc